File: YoutubeLinks.Shared/Exceptions/ErrorResponse.cs
namespace YoutubeLinks.Shared.Exceptions;

public class ErrorResponse(ExceptionType type, string message)
{
    public ExceptionType Type { get; set; } = type;
    public string Message { get; set; } = message;
}

public class ValidationErrorResponse(
    ExceptionType type,
    string message,
    Dictionary<string, List<string>> errors) : ErrorResponse(type, message)
{
    public Dictionary<string, List<string>> Errors { get; set; } = errors;
}

public class ServerErrorResponse(ExceptionType type, string message) : ErrorResponse(type, message) { }

public class UnauthorizedErrorResponse(ExceptionType type, string message) : ErrorResponse(type, message) { }

public class ForbiddenErrorResponse(ExceptionType type, string message) : ErrorResponse(type, message) { }

public class NotFoundErrorResponse(ExceptionType type, string message) : ErrorResponse(type, message) { }

File: youtube-links/src/shared/exceptions/ErrorResponse.ts
import { ExceptionType } from "./ExceptionType";

export interface ErrorResponse {
    type: ExceptionType;
    message: string;
}

export interface ValidationErrorResponse extends ErrorResponse {
    errors: { [key: string]: string[]; };
}

export interface ServerErrorResponse extends ErrorResponse {

}

export interface UnauthorizedErrorResponse extends ErrorResponse {

}

export interface ForbiddenErrorResponse extends ErrorResponse {

}

export interface NotFoundErrorResponse extends ErrorResponse {

}